src/ocean/util/app/ext/TaskExt.d

```diff
@@ -0,0 +1,203 @@
+
+    Copyright:
+        Copyright (c) 2017 sociomantic labs GmbH.
+        All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.util.app.ext.TaskExt;
+
+import ocean.transition;
+import ocean.util.app.ext.model.IConfigExtExtension;
+import ocean.task.Scheduler;
+import ocean.task.Task;
+import ocean.util.config.ConfigParser;
+import ocean.util.config.ConfigFiller;
+import ocean.meta.codegen.Identifier;
+
+/// ditto
+class TaskExt : IConfigExtExtension
+{
+    /// Added to the configuration parsed from the config file if present
+    private SchedulerConfiguration.PoolDescription[] optional_config;
+
+    /// See `enabled`
+    private bool enabled_ = false;
+
+    /// Indicates that automatic scheduler configuration is enabled
+    public bool enabled ( )
+    {
+        return this.enabled_;
+    }
+
+    /***************************************************************************
+
+        Used to enable automatic handling of scheduler configuration and task
+        startup. Designed to be opt-in method to avoid any accidental breakage.
+
+        Params:
+            optional_config = if non-null, will be used as part of scheduler
+                configuration to init it
+
+    ***************************************************************************/
+
+    public void enable ( typeof(SchedulerConfiguration.specialized_pools)
+            optional_config = null )
+    {
+        this.optional_config = optional_config;
+        this.enabled_ = true;
+    }
+
+    /***************************************************************************
+
+        Parse the configuration file options to get the scheduler configuration
+
+        Params:
+            app = the application instance
+            parser = configuration instance
+
+    ***************************************************************************/
+
+    public override void processConfig ( IApplication app, ConfigParser parser )
+    {
+        if (!this.enabled)
+            return;
+
+        static struct PartialSchedulerConfiguration
+        {
+            size_t worker_fiber_stack_size =
+                SchedulerConfiguration.worker_fiber_stack_size.init;
+            size_t worker_fiber_limit =
+                SchedulerConfiguration.worker_fiber_limit.init;
+            size_t task_queue_limit =
+                SchedulerConfiguration.task_queue_limit.init;
+            size_t suspended_task_limit =
+                SchedulerConfiguration.suspended_task_limit.init;
+        }
+
+        // can't fill SchedulerConfiguration directly because it contains
+        // `PoolDescription` array which has to be provided via code
```

Comment:
One idea that just came to my mind is that we can use fully-qualified class name which is converted into `ClassInfo` later internally. That way it can be stored in config file.

Reply:
Ahh I was wondering if there's a runtime way to go from a string to a `ClassInfo`. Would be cool if there is.

Reply:
There is https://github.com/sociomantic-tsunami/tangort/blob/32ad03f41da038b6e315931c551ddfe4cd0d26eb/src/rt/include/object.di#L102 but it has same limitation as all runtime reflection facilities - it completely ignores all templated classes.

Reply:
Oof. Even an alias to a templated class?

Reply:
Yep. Only way to make it work is to make a dummy non-templated class that inherits from a templated one.

Reply:
Ok so there's at least a workaround. Do you think the trouble (fully qualified names / potential workarounds required) is worth it?

Reply:
Not yet sure. Being able to adjust fiber stack sizes via config file sounds quite desirable when you think about something like connection handler in propulsor. But introducing brand new feature with such inherent flaw is a bit awkward.